mula/scheduler/models/tasks.py

```diff
@@ -57,10 +57,39 @@ class Task(BaseModel):
 
     status: TaskStatus
 
-    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
+    meta: dict | None = None
+
+    # Status transition timestamps
+    pending_at: datetime | None = Field(default_factory=lambda: datetime.now(timezone.utc))
```

Comment:
This is just the same as created_at? Could we reduce this to:

`created_at`
`queued_at`
`started_at`
`ended_at`

The latter two being set by the runner?

Reply:
This way you know how long:
- it takes for tasks to complete from queueing them (ended_at - queued_at)
- When the task actually started (started_at)
- The runtime (ended_at - started_at)
- The complete lifecycle (ended_at - created_at) for throughput analysis.
- Queueing time (started_at - queued_at (?)) 

Also, I'd imagine for longrunning tasks that you'd be interested in when they started, not when they were queued. That's I'd say a valid use-case for the started_at/ended_at approach vs. the runtimes? Unless you want to infer that through modified_at. Or the end times in general would be a relevant column to analyse

Reply:
What I just thought of: I guess the default sorting is going to be the `modified_at` in the setup where we only save runtimes?